src/main/java/com/bitwig/extensions/controllers/arturia/keylab/mk2/ArturiaKeylabMkII.java

```diff
@@ -506,6 +511,87 @@ private void initMultiLayer()
       });
    }
 
+   private void initNotificationLayer()
+   {
+      mNotificationLayer.showText(this::getNotificationTopLine, this::getNotificationBottomLine);
+
+      for (final RelativeHardwareControl encoder : mEncoders)
+      {
+         encoder.isUpdatingTargetValue().markInterested();
+         encoder.targetName().markInterested();
+         encoder.targetDisplayedValue().markInterested();
+         encoder.isBeingTouched().markInterested();
+         encoder.targetValue().addValueObserver((v) -> {
+            if (encoder.isUpdatingTargetValue().get())
+               showNotificationOnDisplay(encoder.targetName(), encoder.targetDisplayedValue());
+         });
+      }
+
+      for (final AbsoluteHardwareControl fader : mFaders)
+      {
+         fader.targetName().markInterested();
+         fader.targetDisplayedValue().markInterested();
+         fader.value().addValueObserver((v) -> showNotificationOnDisplay(fader.targetName(), fader.targetDisplayedValue()));
+      }
+   }
+
+   private void showNotificationOnDisplay(final StringValue topRowText, final StringValue bottomRowText)
+   {
+      final int notificationDurationInMs = 500;
```

Comment:
One minor change. Maybe we can instead just have a pointer to the ConttinuousHardwareControl that we are adjusting rather than having a top and bottom row text?